### PR TITLE
chore: update pnpm 8

### PR DESCRIPTION
**Pull request by dominikg**

### CODE REVIEW

**Comment by dominikg** [comment]
looks like the file: protocol in for _test_dependencies works differently in pnpm@8 :/

**Review by bluwy — APPROVED** [review]

**Comment by bluwy** [comment]
Ah wait I approved without reading the comments 🙈

**Comment by benmccann** [comment]
Should we use `pnpm/action-setup` in the CI like SvelteKit does? I noticed this repo has some custom logic currently which could be simplified

File: package.json [hunk 1]
```diff
@@ -55,11 +55,8 @@
       "prettier --write"
     ]
   },
-  "packageManager": "pnpm@7.30.0",
+  "packageManager": "pnpm@8.2.0",
   "engines": {
-    "pnpm": "^7.13.0",
-    "yarn": "forbidden, use pnpm",
-    "npm": "forbidden, use pnpm",
```
Comment (benmccann) [inline package.json:62]: you can leave this in place if you'd like. you just need to switch `^` to `>=` which I believe is the thing that's currently stopping me from using this repo

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -87,11 +87,15 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
       - name: install pnpm
+        if: matrix.node != 14
         shell: bash
         run: |
           PNPM_VER=$(jq -r '.packageManager | if .[0:5] == "pnpm@" then .[5:] else "packageManager in package.json does not start with pnpm@\n" | halt_error(1)  end' package.json)
           echo installing pnpm version $PNPM_VER
           npm i -g pnpm@$PNPM_VER
+      - name: install legacy pnpm for node14
```
Comment (benmccann) [inline .github/workflows/ci.yml:96]: you could just leave `packageManager` as `7.32.0` in `package.json` and then you wouldn't need to do any of this. that's much simpler. people can still have pnpm 8 in their local installs, which is what's really important

**Comment by dominikg** [comment]
> Should we use `pnpm/action-setup` in the CI like SvelteKit does? I noticed this repo has some custom logic currently which could be simplified

that action uses curlbashing and it also adds an additional point where pnpm version is set. this setup here uses packageManager as canonical and doesnt require a separate action. ymmv but i very much prefer the current approach

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dominikg) [reply .github/workflows/ci.yml:96]: the goal is that everyone uses the same major version of the same pm. differences between pnpm 7 and 8 make it undesirable to allow contributing with both, as evidenced by the current problem of file: resolution in this PRs test run

File: package.json (hunk 1 above)
Comment (dominikg) [reply package.json:62]: you can enable corepack locally to ease working with different package manager versions, or switch manually. until the issue with 8 is resolved, 7 must be used.

**Comment by benmccann** [comment]
Just FYI, the action also uses `packageManager` and doesn't require setting the version used separately from that

File: package.json (hunk 1 above)
Comment (dominikg) [reply package.json:62]: added back engines.pnpm and rewriting it for node14 tests. left the other two removed as neither npm nor yarn check it anymore i think.

**Review by bluwy — APPROVED** [review]

**Pull request merged**